src/main/java/uk/gov/companieshouse/web/pps/util/PenaltyUtils.java

```diff
@@ -27,5 +29,18 @@ public String getFormattedOutstanding(final Integer outstandingAmount) {
     public String getReferenceTitle(final String penaltyNumber) {
         return penaltyNumber.startsWith("A") ? "Reference Number" : "Penalty Reference";
     }
+
+    public String getLoginEmail(SessionService sessionService) {
+        Map<String, Object> sessionData = sessionService.getSessionDataFromContext();
+        Map<String, Object> signInInfo = (Map<String, Object>) sessionData.get("signin_info");
+        if (signInInfo != null) {
+            Map<String, Object> userProfile = (Map<String, Object>) signInInfo
+                    .get("user_profile");
+            if (userProfile != null) {
+                return userProfile.get("email").toString();
+            }
+        }
+        return null;
```

Comment:
I am not a fan of returning nulls. I know we are doing it in other places in our new code but I think that this is wrong. java 8 introduced Optionals so this could be avoided.